[BUG] NPE in OTel sources during initialization

**Describe the bug**
OTel sources are throwing an IllegalArgumentException because of the code change introduced in https://github.com/opensearch-project/data-prepper/pull/2297

pipelineName here is null as it's being initialized after the call to this method. https://github.com/opensearch-project/data-prepper/blob/437fd72378fbb8011e6167715ef195aa40a96f3b/data-prepper-plugins/otel-logs-source/src/main/java/com/amazon/dataprepper/plugins/source/otellogs/OTelLogsSource.java#L205

https://github.com/opensearch-project/data-prepper/blob/437fd72378fbb8011e6167715ef195aa40a96f3b/data-prepper-plugins/otel-logs-source/src/main/java/com/amazon/dataprepper/plugins/source/otellogs/OTelLogsSource.java#L73-L74

**Solution**
Above code (line 205) change should be reverted back to use 
```
pipelineDescription.getPipelineName();
```



**To Reproduce**
Steps to reproduce the behavior:
Create a pipeline with http_basic auth in any of the 3 OTel source.

**Expected behavior**
Caused by: java.lang.IllegalArgumentException: PluginSetting.pipelineName must not be null


**Environment (please complete the following information):**
 - OS: [e.g. Ubuntu 20.04 LTS]
 - Version [e.g. 22]

**Additional context**
Add any other context about the problem here.

--- a/data-prepper-plugins/otel-logs-source/src/main/java/com/amazon/dataprepper/plugins/source/otellogs/OTelLogsSource.java
+++ b/data-prepper-plugins/otel-logs-source/src/main/java/com/amazon/dataprepper/plugins/source/otellogs/OTelLogsSource.java
@@ -70,8 +70,8 @@ public class OTelLogsSource implements Source<Record<Object>> {
         this.oTelLogsSourceConfig = oTelLogsSourceConfig;
         this.pluginMetrics = pluginMetrics;
         this.certificateProviderFactory = certificateProviderFactory;
-        this.authenticationProvider = createAuthenticationProvider(pluginFactory, pipelineDescription);
         this.pipelineName = pipelineDescription.getPipelineName();
+        this.authenticationProvider = createAuthenticationProvider(pluginFactory);
     }
 
     @Override
@@ -188,7 +188,7 @@ public class OTelLogsSource implements Source<Record<Object>> {
         return Collections.singletonList(authenticationInterceptor);
     }
 
-    private GrpcAuthenticationProvider createAuthenticationProvider(final PluginFactory pluginFactory, final PipelineDescription pipelineDescription) {
+    private GrpcAuthenticationProvider createAuthenticationProvider(final PluginFactory pluginFactory) {
         final PluginModel authenticationConfiguration = oTelLogsSourceConfig.getAuthentication();
 
         if (authenticationConfiguration == null || authenticationConfiguration.getPluginName().equals(GrpcAuthenticationProvider.UNAUTHENTICATED_PLUGIN_NAME)) {
@@ -202,7 +202,7 @@ public class OTelLogsSource implements Source<Record<Object>> {
         } else {
             authenticationPluginSetting = new PluginSetting(GrpcAuthenticationProvider.UNAUTHENTICATED_PLUGIN_NAME, Collections.emptyMap());
         }
-        authenticationPluginSetting.setPipelineName(pipelineDescription.getPipelineName());
+        authenticationPluginSetting.setPipelineName(pipelineName);
         return pluginFactory.loadPlugin(GrpcAuthenticationProvider.class, authenticationPluginSetting);
     }
 }

--- a/data-prepper-plugins/otel-logs-source/src/main/java/com/amazon/dataprepper/plugins/source/otellogs/OTelLogsSource.java
+++ b/data-prepper-plugins/otel-logs-source/src/main/java/com/amazon/dataprepper/plugins/source/otellogs/OTelLogsSource.java
@@ -202,7 +202,7 @@ public class OTelLogsSource implements Source<Record<Object>> {
         } else {
             authenticationPluginSetting = new PluginSetting(GrpcAuthenticationProvider.UNAUTHENTICATED_PLUGIN_NAME, Collections.emptyMap());
         }
-        authenticationPluginSetting.setPipelineName(pipelineName);
+        authenticationPluginSetting.setPipelineName(pipelineDescription.getPipelineName());
         return pluginFactory.loadPlugin(GrpcAuthenticationProvider.class, authenticationPluginSetting);
     }
 }

--- a/data-prepper-plugins/otel-metrics-source/src/main/java/org/opensearch/dataprepper/plugins/source/otelmetrics/OTelMetricsSource.java
+++ b/data-prepper-plugins/otel-metrics-source/src/main/java/org/opensearch/dataprepper/plugins/source/otelmetrics/OTelMetricsSource.java
@@ -220,7 +220,7 @@ public class OTelMetricsSource implements Source<Record<ExportMetricsServiceRequ
         } else {
             authenticationPluginSetting = new PluginSetting(GrpcAuthenticationProvider.UNAUTHENTICATED_PLUGIN_NAME, Collections.emptyMap());
         }
-        authenticationPluginSetting.setPipelineName(pipelineName);
+        authenticationPluginSetting.setPipelineName(pipelineDescription.getPipelineName());
         return pluginFactory.loadPlugin(GrpcAuthenticationProvider.class, authenticationPluginSetting);
     }
 }

--- a/data-prepper-plugins/otel-metrics-source/src/main/java/org/opensearch/dataprepper/plugins/source/otelmetrics/OTelMetricsSource.java
+++ b/data-prepper-plugins/otel-metrics-source/src/main/java/org/opensearch/dataprepper/plugins/source/otelmetrics/OTelMetricsSource.java
@@ -72,8 +72,8 @@ public class OTelMetricsSource implements Source<Record<ExportMetricsServiceRequ
         this.oTelMetricsSourceConfig = oTelMetricsSourceConfig;
         this.pluginMetrics = pluginMetrics;
         this.certificateProviderFactory = certificateProviderFactory;
-        this.authenticationProvider = createAuthenticationProvider(pluginFactory, pipelineDescription);
         this.pipelineName = pipelineDescription.getPipelineName();
+        this.authenticationProvider = createAuthenticationProvider(pluginFactory);
     }
 
     @Override
@@ -206,7 +206,7 @@ public class OTelMetricsSource implements Source<Record<ExportMetricsServiceRequ
         return Collections.singletonList(authenticationInterceptor);
     }
 
-    private GrpcAuthenticationProvider createAuthenticationProvider(final PluginFactory pluginFactory, final PipelineDescription pipelineDescription) {
+    private GrpcAuthenticationProvider createAuthenticationProvider(final PluginFactory pluginFactory) {
         final PluginModel authenticationConfiguration = oTelMetricsSourceConfig.getAuthentication();
 
         if (authenticationConfiguration == null || authenticationConfiguration.getPluginName().equals(GrpcAuthenticationProvider.UNAUTHENTICATED_PLUGIN_NAME)) {
@@ -220,7 +220,7 @@ public class OTelMetricsSource implements Source<Record<ExportMetricsServiceRequ
         } else {
             authenticationPluginSetting = new PluginSetting(GrpcAuthenticationProvider.UNAUTHENTICATED_PLUGIN_NAME, Collections.emptyMap());
         }
-        authenticationPluginSetting.setPipelineName(pipelineDescription.getPipelineName());
+        authenticationPluginSetting.setPipelineName(pipelineName);
         return pluginFactory.loadPlugin(GrpcAuthenticationProvider.class, authenticationPluginSetting);
     }
 }

--- a/data-prepper-plugins/otel-trace-source/src/main/java/org/opensearch/dataprepper/plugins/source/oteltrace/OTelTraceSource.java
+++ b/data-prepper-plugins/otel-trace-source/src/main/java/org/opensearch/dataprepper/plugins/source/oteltrace/OTelTraceSource.java
@@ -73,8 +73,8 @@ public class OTelTraceSource implements Source<Record<Object>> {
         this.oTelTraceSourceConfig = oTelTraceSourceConfig;
         this.pluginMetrics = pluginMetrics;
         this.certificateProviderFactory = certificateProviderFactory;
-        this.authenticationProvider = createAuthenticationProvider(pluginFactory, pipelineDescription);
         this.pipelineName = pipelineDescription.getPipelineName();
+        this.authenticationProvider = createAuthenticationProvider(pluginFactory);
     }
 
     @Override
@@ -210,7 +210,7 @@ public class OTelTraceSource implements Source<Record<Object>> {
         return Collections.singletonList(authenticationInterceptor);
     }
 
-    private GrpcAuthenticationProvider createAuthenticationProvider(final PluginFactory pluginFactory, final PipelineDescription pipelineDescription) {
+    private GrpcAuthenticationProvider createAuthenticationProvider(final PluginFactory pluginFactory) {
         final PluginModel authenticationConfiguration = oTelTraceSourceConfig.getAuthentication();
 
         if (authenticationConfiguration == null || authenticationConfiguration.getPluginName().equals(GrpcAuthenticationProvider.UNAUTHENTICATED_PLUGIN_NAME)) {
@@ -224,7 +224,7 @@ public class OTelTraceSource implements Source<Record<Object>> {
         } else {
             authenticationPluginSetting = new PluginSetting(GrpcAuthenticationProvider.UNAUTHENTICATED_PLUGIN_NAME, Collections.emptyMap());
         }
-        authenticationPluginSetting.setPipelineName(pipelineDescription.getPipelineName());
+        authenticationPluginSetting.setPipelineName(pipelineName);
         return pluginFactory.loadPlugin(GrpcAuthenticationProvider.class, authenticationPluginSetting);
     }
 }

--- a/data-prepper-plugins/otel-trace-source/src/main/java/org/opensearch/dataprepper/plugins/source/oteltrace/OTelTraceSource.java
+++ b/data-prepper-plugins/otel-trace-source/src/main/java/org/opensearch/dataprepper/plugins/source/oteltrace/OTelTraceSource.java
@@ -224,7 +224,7 @@ public class OTelTraceSource implements Source<Record<Object>> {
         } else {
             authenticationPluginSetting = new PluginSetting(GrpcAuthenticationProvider.UNAUTHENTICATED_PLUGIN_NAME, Collections.emptyMap());
         }
-        authenticationPluginSetting.setPipelineName(pipelineName);
+        authenticationPluginSetting.setPipelineName(pipelineDescription.getPipelineName());
         return pluginFactory.loadPlugin(GrpcAuthenticationProvider.class, authenticationPluginSetting);
     }
 }

